bigtable/tests/data_integration_test.cc

```diff
@@ -12,12 +12,74 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <gmock/gmock.h>
+
+#include <absl/memory/memory.h>
+#include <absl/strings/str_join.h>
+
 #include "bigtable/admin/admin_client.h"
 #include "bigtable/admin/table_admin.h"
+#include "bigtable/client/cell.h"
+#include "bigtable/client/data_client.h"
 #include "bigtable/client/table.h"
 
+namespace admin_proto = ::google::bigtable::admin::v2;
+
+namespace {
+/// Store the project and instance captured from the command-line arguments.
+class DataTestEnvironment : public ::testing::Environment {
+ public:
+  DataTestEnvironment(std::string project, std::string instance,
+                      std::string table_name) {
+    project_id_ = std::move(project);
+    instance_id_ = std::move(instance);
+    table_name_ = std::move(table_name);
+  }
+
+  static std::string const& project_id() { return project_id_; }
+  static std::string const& instance_id() { return instance_id_; }
+  static std::string const& table_name() { return table_name_; }
+
+ private:
+  static std::string project_id_;
+  static std::string instance_id_;
+  static std::string table_name_;
+};
+
+class DataIntegrationTest : public ::testing::Test {
+ protected:
+  void SetUp() override;
+
+  /**
+   * creates the table passed via arguments.
+   */
+  std::unique_ptr<bigtable::Table> CreateTable();
+
+  /**
+  * deletes the table passed via arguments.
+  */
+  void DeleteTable();
```

Comment:
I cannot find the implementation for this function, is it missing?

Reply:
line 236 in data_integration_test, this will be part of re-factored class.

Reply:
Doh!  Thanks, I have updated my review based on that.